proposed/2021/MachineReadableOutput.md

```diff
@@ -0,0 +1,821 @@
+# Machine readable JSON output for dotnet list package
+
+* Status: **In Review**
+* Author: [Erick Yondon](https://github.com/erdembayar)
+* GitHub Issue [7752](https://github.com/NuGet/Home/issues/7752)
+
+## Problem background
+<!-- Why are we doing this? What pain points does this solve? What is the expected outcome? -->
+Many organizations are required by [regulation](https://www.whitehouse.gov/briefing-room/presidential-actions/2021/05/12/executive-order-on-improving-the-nations-cybersecurity/) to audit packages that they're using in a repository.
+
+Currently there's no easy way to produce a [Software Bill of Material (SBOM)](https://blog.sonatype.com/what-is-a-software-bill-of-materials) output which can be consumed by another auditing system or kept for records.
+
+* Parse-friendly output. Other package managers like [NPM already have it](https://docs.npmjs.com/cli/v7/commands/npm-ls) (`npm ls --parseable` and `npm ls --json`).
+* Useful for CI/CD auditing(compliance, security ..)
+  * Produce Software Bill of Material (SBOM) (compliance, historic keeping)
+  * Enhancing Software Supply Chain Security
+    * Check any vulnerable packages
+    * Check any deprecated packages
+    * Check any outdated packages
+  * Check license compliance
+  * Resolve dependency issues (detect duplicate packages, newly introduced dependencies, dependency removals, etc.)
+  * Making the output machine-readable unlocks additional tooling and automation scenarios in CI/CD pipeline above scenarios.
+
+## Who are the customers
+
+Anyone (government/private enterprises, security experts, individual contributors) who wants to consume `dotnet list package` output for auditing tool or historic keeping, CI/CD orchestrating.
+
+## Solution
+
+In order to solve problem going to introduce 2 new options.
+
+1. `--format` option for all `dotnet list package` commands to ensure formatted(json, text, csv etc) output is emitted to the console.
+
+1. `--all` option which works in conjunction with above for that shows all deprecated, vulnerable, and outdated top level and transitive packages. Users can form their own json as a last resort, instead of your team having to respond to every possible user need.
```

Comment:
I think this is scope creep, and should not be a part of this spec. 

The intention of --outdated  conflicts to that of --vulnerable and --deprecated. 

https://github.com/NuGet/Home/wiki/Deprecate-packages#dotnet-list-package---outdated---deprecated

Reply:
Reverted.